### PR TITLE
Python wrappers

### PR DESCRIPTION
**Pull request by mortezah**
This adds minimum required wrappers to be able to call SCOREC adaptive and curving tools from within python. This is achieved using SWIG.

### CODE REVIEW

File: python_wrappers/cmake/FindSimModSuite.cmake [hunk 1]
```diff
@@ -0,0 +1,182 @@
+# - Try to find Simmetrix SimModSuite
```
Comment (cwsmith) [inline python_wrappers/cmake/FindSimModSuite.cmake]: Is this a duplicate of the top-level `FindSimModSuite.cmake` file?

File: python_wrappers/cmake/bob.cmake [hunk 2]
```diff
@@ -0,0 +1,207 @@
+function(bob_always_full_rpath)
```
Comment (cwsmith) [inline python_wrappers/cmake/bob.cmake]: Is this a duplicate of the top-level `bob.cmake` file?

**Comment by mortezah** [comment]
Thanks, @cwsmith for the comments.

I plan to incorporate the creation of _pyCore.so into the top-level build, but before doing so I need to make a few other changes (and I don't know how long the will take). Meanwhile, it would be good if we can have this merged so that Syunidchi can use them. Having things in two different branches makes his workflow a bit tedious, so I thought I could just merge a working version to the develop branch.

I will fix the redundancies in regards to the cmake stuff and let you know when I am done.

**Comment by cwsmith** [comment]
I'll take a look at the top-level cmake integration.  If this needs to shared ASAP we can create a fork and periodically rebase it off develop/master.

**Comment by mortezah** [comment]
Just a quick note, the _pyCore.so need to be linked to the .so version of core libraries.

**Comment by cwsmith** [comment]
OK.  Thanks for the heads up.  We can force dynamic when python wrappers are enabled.

**Comment by mortezah** [comment]
@cwsmith I took care of repeated FindModule and bob.cmake. 

Did you have any luck with including the wrapper build into the higher-level build?

**Comment by cwsmith** [comment]
I made some progress:
https://github.com/SCOREC/core/tree/pyWrapTopLvl
The current issue is that `swig_link_libraries(...)` does not add include directories of the `core` target.  According to the docs (https://cmake.org/cmake/help/v3.12/module/UseSWIG.html) it is supposed to function identically to `target_link_libraries(...)` which will add include dirs and libraries.

Using https://cmake.org/cmake/help/v3.12/module/FindPython3.html#module:FindPython3 worked without issue.  On scorec systems our mpi4py was built with python2 so I used teh corresponding `find_package(Python2)`.  Once things are stable we can support both versions if needed.

**Comment by cwsmith** [comment]
@mortezah This appears to work:

https://github.com/SCOREC/core/commit/b6d491499a2e7b4bd33a9b4c38f6beeecb5794b4

My build process on scorec is:

```
module load petram
source $TwoPiRoot/bin/activate
module load python/2.7.5-bbskhpf
cmake ../core -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx -DCMAKE_Fortran_COMPILER=gfortran -DSCOREC_CXX_OPTIMIZE=on -DSCOREC_EXTRA_CXX_FLAGS= -DVALGRIND= -DVALGRIND_ARGS= -D
MDS_ID_TYPE=int -DPCU_COMPRESS=ON -DENABLE_ZOLTAN=off -DPUMI_FORTRAN_INTERFACE=off -DPUMI_PYTHON_INTERFACE=on -DBUILD_SHARED_LIBS=on -DBUILD_EXES=off -DENABLE_OMEGA_H=off -DIS_TESTING=O
N -DENABLE_SIMMETRIX=off -DENABLE_FIELDSIM=OFF -DSIM_MPI=mpich3.1.2 -DSIM_PARASOLID=off -DSIM_ACIS=off -DMESHES=/lore/cwsmith/geometries/pumi-meshes/ -DCMAKE_INSTALL_PREFIX=/lore/cwsmit
h/develop/build-core-rhel7-python/install
```

**Comment by mortezah** [comment]
Thanks @cwsmith 

I will test it out and let you know.

**Comment by mortezah** [comment]
@cwsmith  I just finished up with including build with simmetrix options, and tested it out for both cases (i.e., with and without simmetrix).

Let me know if you need me to make any other changes.

**Pull request merged**